### ISSUE TITLE
Best integration of oft / ttf math fonts in Manim

### ISSUE THREAD

**Issue by jxm-math** [3b1b/manim#858]
By changing the preamble in `tex_template.tex`, one may use different fonts for the animations, both for text and math formulas. However, [some of the most interesting math fonts](https://tug.org/FontCatalogue/mathfonts.html) are only available in otf / ttf, which is usually related with a xelatex engine. 

Is it possible right now to use these fonts in Manim? Which is the easiest integration?

Thanks in advance

**Comment by zavden** [comment]
https://github.com/3b1b/manim/pull/680

**Comment by jxm-math** [comment]
Does it support math formulas?

**Comment by zavden** [comment]
No, if you want to use the Math format you have to add the corresponding package to the tex_template.tex file, if you want to create multiple files tex_template I recommend you see [this PR](https://github.com/3b1b/manim/pull/745), for example, if you want to add [this font](https://tug.org/FontCatalogue/antykwatorunskacondensed/) then you must add this to your tex_template file:
```latex
\usepackage[condensed,math]{anttor}
\usepackage[T1]{fontenc}
```
And use it like:
```python
TextMobject(r"\normalfont Normal font \scshape SC shape")
```
This only works if you have the font installed, if not, you must install it, either with the help of TexMaker or manually. If you do it manually, you must download the source from its official repository in the CTAN ([this is the repository](https://www.ctan.org/tex-archive/fonts/antt/) of the example font) and install it. If you don't know how to do that you should investigate it on your own.

**Comment by kexplorning** [comment]
@jxm-math for XeLaTeX, here is a workaround (a dirty trick). Set `TEX_USE_CTEX=True` in `constants.py` (because `ctex` needs XeLaTeX to work), and then edit `ctex_template.tex` like this:

```latex
\usepackage{unicode-math}
\setmathfont{<math font>}
```

**Comment by kexplorning** [comment]
@zavden This method only works with the tfm font (the old TeX font), if you want to include modern font (ttf and oft font), you need XeTeX or LuaTeX.

**Comment by jxm-math** [comment]
@kexplorning the trick works great. I added

```
\setmainfont{Lato-Light.otf}

\newfontfamily\texttt[%
  Scale=MatchLowercase
]{SourceCodePro-ExtraLight.otf}

\newfontfamily\fira[%
  Scale=MatchLowercase
]{FiraSans-ExtraLight.otf}

\RequirePackage{unicode-math}
\setmathfont[%
       Scale=1.2%use instead of 12pt 
]{GFSNeohellenicMath.otf}
```

 to `ctex_template.tex`, then set `TEX_USE_CTEX=True` and installed the required libraries.

You may see the result [here](https://twitter.com/74WTungsteno/status/1223298968106479622?s=20): math font is GFSNeohellenicMath

Additionally, here's the code of the video: https://github.com/TungstenHub/tngt-manim

Thanks @zavden too for your very useful comments
